lib/devfest_registration_portal_web/views/challenge_view.ex

```diff
@@ -1,3 +1,12 @@
 defmodule DevfestRegistrationPortalWeb.ChallengeView do
   use DevfestRegistrationPortalWeb, :view
+  alias DevfestRegistrationPortal.Codelabs
+
+  def category_name do
+    categories = Codelabs.list_all_categories()
```

Comment:
instead of calling `list_all_categories/0` from the view, i would suggest you do it from the controller then pass the data to `assigns`.

Reply:
in my opinion views should just be processing existing data, not querying for the data